chord_metadata_service/chord/ingest/views.py

```diff
@@ -26,18 +26,59 @@
 logger = logging.getLogger(__name__)
 
 
+class IngestResponseBuilder:
+
+    def __init__(self, workflow_id: str, dataset_id: str):
+        self.workflow_id = workflow_id
+        self.dataset_id = dataset_id
+        self.success = False
+        self.errors = []
+        self.warnings = []
+
+    def set_success(self, success: bool):
+        self.success = success
+
+    def add_error(self, error):
+        self.errors.append(error)
+
+    def add_errors(self, errors: List):
+        self.errors.extend(errors)
+
+    def add_ingest_error(self, error: IngestError):
+        if error.validation_errors:
+            self.add_errors(error.validation_errors)
+        else:
+            self.add_error(error.message)
+
+        if error.schema_warnings:
+            self.warnings.extend(error.schema_warnings)
+
+    def as_response(self, status_code: int) -> Response:
+        body = {
+            "success": self.success,
```

Comment:
rather than needing to manually mark success, this could be derived from if the status code is <400

Reply:
good point